File: 卡码/代码随想录/1/09-动态规划/Python/02-背包问题/02-完全背包/LeetCode_0518.py
#518. 零钱兑换 II
from typing import *

class Solution:
  def change(self, amount: int, coins: List[int]) -> int:
    dp:List[int]=[0]*(amount+1)
    dp[0]=1
    for i in range(len(coins)):
      for j in range(coins[i],amount+1):
        dp[j]+=dp[j-coins[i]]
    return dp[amount]

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/动态规划/_02_背包问题/_02_完全背包/LeetCode_0518.java
package com.kurumi.leetCode.codeThoughts.动态规划._02_背包问题._02_完全背包;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 518. 零钱兑换 II
 */
public class LeetCode_0518 {
    static class Solution {
        public int change(int amount, int[] coins) {
            int[] dp = new int[amount + 1];
            dp[0] = 1;
            for (int i = 0; i < coins.length; i++) {
                for (int j = coins[i]; j <= amount; j++) {
                    dp[j] += dp[j - coins[i]];
                }
            }
            return dp[amount];
        }
    }
}
